tensorflow/core/kernels/cwise_ops_gradients.h

```diff
@@ -75,14 +75,19 @@ struct scalar_inverse_gradient_op {
   EIGEN_EMPTY_STRUCT_CTOR(scalar_inverse_gradient_op)
   EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const T
   operator()(const T& output, const T& output_gradient) const {
-    const T out_conj = numext::conj(output);
-    return -output_gradient * out_conj * out_conj;
+    if (output_gradient == T(0)) {
+      return T(0);
+    } else {
+      const T out_conj = numext::conj(output);
+      return -out_conj * out_conj * output_gradient;
+    }
   }
   template <typename Packet>
   EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const Packet
   packetOp(const Packet& output, const Packet& output_gradient) const {
     const Packet out_conj = pconj(output);
-    return pnegate(pmul(output_gradient, pmul(out_conj, out_conj)));
+    return mul_no_nan_op<T>().packetOp(pnegate(pmul(out_conj, out_conj)),
```

Comment:
I think you will need to set PacketAccess to False for TENSORFLOW_USE_ROCM at line 97 to get around the error you are seeing. Doing so should disable this routine from getting invoked.